hazelcast/src/main/java/com/hazelcast/map/impl/query/MapQueryEngineImpl.java

```diff
@@ -275,6 +280,11 @@ public QueryResult newQueryResult(int numberOfPartitions) {
         return new QueryResult(queryResultSizeLimiter.getNodeResultLimit(numberOfPartitions));
     }
 
+    @Override
+    public Predicate optimize(Predicate predicate, Indexes indexes) {
```

Comment:
I guess it is caused by the call the 'indexes' ? The indexes should be behind the QueryEngine.